region/maridia/outer/Crab Hole.json

```diff
@@ -1047,6 +1047,276 @@
+              "maxHeight": 1,
+              "minTiles": 17,
+              "speedBooster": true,
+              "obstructions": [[1, 0]],
+              "requires": [
+                {"or": [
+                  "canMomentumConservingMorph",
+                  {"and": [
+                    "canMomentumConservingTurnaround",
+                    "canTrickyJump"
+                  ]}
+                ]}
+              ],
+              "note": [
+                "This applies to Warehouse Entrance.",
+                "This can be done in three ways:",
+                "1) Jump on the last possible frame in the previous room, turn around a little before hitting the ceiling, then morph and hold right to pass through the tunnel.",
+                "2) Perform a ceiling mockball through the transition,",
+                "unmorph a little before hitting the ceiling,",
+                "hold right to press against the wall (to avoid hitting the crab), then morph and hold right to pass through the tunnel.",
+                "3) Jump on the last possible frame in the previous room, then perform a ceiling mockball to go up through the hidden space,",
+                "briefly releasing right while approaching the ceiling, in order to avoid hitting the crab."
+              ],
+              "devNote": [
+                "The turnaround method has about a 9-frame window for the turnaround.",
+                "The through-door ceiling mockball has about a 4-frame window for the jump, between a 1-frame and a 5-frame window for the morph (with later jumps giving a larger window),",
+                "and about a 7-frame window for the unmorph.",
+                "The after-transition ceiling mockball has an 8-frame for the morph.",
+                "Kagoing the crab is also possible, but it seems more difficult than all three other options, and it takes damage, so is probably pointless."
+              ]
+            },
+            {
+              "minHeight": 2,
+              "maxHeight": 2,
+              "minTiles": 8.4375,
+              "speedBooster": "any",
+              "obstructions": [[1, 0]],
+              "requires": [
+                "canTrickyJump",
+                "canMomentumConservingTurnaround"
+              ],
+              "note": [
+                "This applies to Ridley Tank Room and Dust Torizo Room.",
+                "From Ridley Tank Room, avoid backing into the corner.",
+                "From Dust Torizo Room, it doesn't matter."
+              ],
+              "devNote": [
+                "Speed Booster is not required for this."
+              ]
+            },
+            {
+              "minHeight": 3,
+              "maxHeight": 3,
+              "minTiles": 8.4375,
+              "speedBooster": true,
+              "obstructions": [[1, 0], [3, 2]],
+              "requires": [
+                {"or": [
+                  "canMomentumConservingMorph",
+                  "canMomentumConservingTurnaround"
```

Comment:
For Metal Pirates its `canTrickyDashJump` on top of the momentumeConservingTurnaround. If you aren't on the correct tile window it becomes a much harder jump. 
I don't have a blue door patch for big boy room but it might get a pass for not being heated.

Reply:
Cool, added "canTrickyDashJump" requirement to the "canMomentumConservingTurnaround" case.

If it helps, I shared a blue doors patch in #development (Map Rando discord).